### ISSUE TITLE
False error with bitarray

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#266]
Originally reported by: **Ron Johnson (BitBucket: [RonJohn](http://bitbucket.org/RonJohn), GitHub: @RonJohn?)**

----------------------------------------

Hi,

```
#!text

$ pylint --version
No config file found, using default configuration
pylint 1.2.1, 
astroid 1.1.1, common 0.61.0
Python 2.7.5+ (default, Feb 27 2014, 19:37:08) 
[GCC 4.8.1]

$ apt-cache policy python-bitarray
python-bitarray:
  Installed: 0.8.0-2build2
  Candidate: 0.8.0-2build2
  Version table:
 *** 0.8.0-2build2 0
        500 http://us.archive.ubuntu.com/ubuntu/ saucy/universe amd64 Packages
        100 /var/lib/dpkg/status
```

pylint throws this error when processing a file which uses the setall() method:

```
#!text

E: 18, 0: Instance of 'bitarray' has no 'setall' member (no-member)
```

I know this is an error, because the code works perfectly:

```
#!python

>>> from bitarray import bitarray
>>> a = bitarray(32)
>>> a.setall(True)
>>> print a
bitarray('11111111111111111111111111111111')
>>> a.setall(False)
>>> print a
bitarray('00000000000000000000000000000000')
>>> 
```

Sincerely,
Ron

----------------------------------------
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/266

**Comment by PCManticore** [comment]
This now works somehow on astroid 2.0, where we have a better mechanism for understanding C extensions. I'm saying somehow because due to a bigger problem with inference contexts, we might not infer correctly all the methods of a bitarray instance (the corresponding issue is https://github.com/PyCQA/astroid/issues/108), but at least we don't emit this false positive anymore. A way to test would, at least for now, would be with something like this (where astroid 2.0 is used):

    from astroid.test_utils import extract_node
    from astroid import MANAGER
    MANAGER.always_load_extensions = True
    n = extract_node('''
    from bitarray import bitarray
    b = bitarray(32)
    b
    ''')
    i = next(n.infer())
    print(next(i.igetattr('setall')))

**Comment by PCManticore** [comment]
Forgot to say though that this will only work in Pylint 2.0 with the ``--extension-pkg-whitelist=bitarray._bitarray`` activated.
